source/LocationFilter.cpp

```diff
@@ -164,13 +181,15 @@ void LocationFilter::Save(DataWriter &out) const
 			}
 			out.EndChild();
 		}
-		if(!governments.empty())
+		if(!governments.empty() || !governmentsBlacklist.empty())
 		{
 			out.Write("government");
 			out.BeginChild();
 			{
 				for(const Government *government : governments)
 					out.Write(government->GetName());
+				for(const Government *government : governmentsBlacklist)
```

Comment:
As above, probably only need to write one of these, as they both should not exist for the same mission. Include is more restrictive than exclude, so I would write that one and `else` write the blacklist lines.